test/services/ldap-config/service.integration.test.js

```diff
@@ -0,0 +1,381 @@
+	let app;
+	let server;
+	let configBefore;
+	let testObjects;
+
+	before(async () => {
+		delete require.cache[require.resolve('../../../src/app')];
+		configBefore = Configuration.toObject({ plainSecrets: true });
+		Configuration.set('FEATURE_API_VALIDATION_ENABLED', true);
+		Configuration.set('FEATURE_API_RESPONSE_VALIDATION_ENABLED', true);
+		// eslint-disable-next-line global-require
+		const appPromise = require('../../../src/app');
+		// eslint-disable-next-line global-require
+		testObjects = require('../helpers/testObjects')(appPromise);
+		app = await appPromise;
+		server = await app.listen(0);
+	});
+
+	after(async () => {
+		await testObjects.cleanup();
+		await server.close();
+		Configuration.reset(configBefore);
+	});
+
+	describe('GET route', () => {
+		let generalLdapSystem1;
+		let iservLdapSystem1;
+		let otherSystem1;
+		let system2;
+		let school1;
+		let school2;
+		let admin1;
+		let teacher2;
+		let paramsAdmin1;
+		let paramsTeacher2;
+
+		before(async () => {
+			generalLdapSystem1 = await testObjects.createTestSystem({
+				type: 'ldap',
+				ldapConfig: {
+					url: 'ldaps://foo.bar:636',
+					provider: 'general',
+					providerOptions: {},
+				},
+			});
+			iservLdapSystem1 = await testObjects.createTestSystem({
+				type: 'ldap',
+				ldapConfig: {
+					url: 'ldaps://foo.bar:10636',
+					provider: 'iserv',
+					providerOptions: {},
+				},
+			});
+			otherSystem1 = await testObjects.createTestSystem({
+				type: 'moodle',
+			});
+			school1 = await testObjects.createTestSchool({
+				systems: [generalLdapSystem1._id, iservLdapSystem1._id, otherSystem1._id],
+			});
+			admin1 = await testObjects.createTestUser({
+				schoolId: school1._id,
+				roles: ['administrator'],
+			});
+			paramsAdmin1 = await testObjects.generateRequestParamsFromUser(admin1);
+
+			system2 = await testObjects.createTestSystem({
+				type: 'ldap',
+				ldapConfig: {
+					url: 'ldaps://badumm.tsss',
+					provider: 'general',
+					providerOptions: {},
+				},
+			});
+			school2 = await testObjects.createTestSchool({
+				systems: [system2._id],
+			});
+			teacher2 = await testObjects.createTestUser({
+				schoolId: school2._id,
+				roles: ['teacher'],
+			});
+			paramsTeacher2 = await testObjects.generateRequestParamsFromUser(teacher2);
+		});
+
+		it("should allow accessing the LDAP config of the user's school", async () => {
+			const result = await app.service('legacy/v1/ldap-config').get(generalLdapSystem1._id, paramsAdmin1);
+			// also note that this does not return the system, but the ldapConfig
+			expect(result.provider).to.equal('general');
+			expect(result.url).to.equal('ldaps://foo.bar:636');
+		});
+
+		it('should not allow accessing systems with provider!=general', async () => {
+			try {
+				await app.service('legacy/v1/ldap-config').get(iservLdapSystem1._id, paramsAdmin1);
+				expect.fail('this should not happen');
+			} catch (err) {
+				expect(err).to.be.instanceOf(Forbidden);
+			}
```

Comment:
```suggestion
	expect(app.service('legacy/v1/ldap-config').get(iservLdapSystem1._id, paramsAdmin1)).to.eventually.throw(new Forbidden());
```

Reply:
In order to use this you have to add chaiAsPromised:

`const chaiAsPromised = require('chai-as-promised');
chai.use(chaiAsPromised);`

Reply:
I've simplified the test cases with your suggestion. Thanks!